src/parse.d

```diff
@@ -1100,7 +1100,6 @@ public:
             TemplateParameters* tpl = null;
             if (token.value == TOKlparen)
                 tpl = parseTemplateParameterList();
-            assert(token.value == TOKassign);
             nextToken(); // skip over '='
```

Comment:
I don't understand this fix. The comment here expects the next token to be `=` but we've just removed the assertion that checks this assumption. Doesn't that mean that malformed user input may cause the following code to malfunction? Shouldn't this be turned into a check with a compile error message when the next token isn't `=` instead?